transaction/depot.go

```diff
@@ -93,6 +96,12 @@ type DepotStorage interface {
 	// UpsertDeliveryRequest is used to update a delivery. If no delivery with a given
 	// unique ID exists, it should create it.
 	UpsertDeliveryRequest(tx Delivery) error
+
+	// GetDeliveredDeliveryRequests should return a list of delivered delivery requests older than the given time. Order is not important.
+	GetDeliveredDeliveryRequests(chainID int64, sender common.Address, olderThan time.Time, limit uint64) ([]Delivery, error)
```

Comment:
Could we make this a separate interface and also optional? My reasoning here is quite simple, maybe I don't want to cleanup any entries, why would I bother implementing this part of the interface or mocking up the methods. The cleaner for me seems more or less like an optional part of the depot and the interface is already pretty large.